### PR TITLE
[expo-notification] Adds DailyTrigger

### PR DESCRIPTION
**Pull request by thorbenprimke**
# Why

In order to trigger a local notification once per day, the `DailyIntervalTriggerInput` is being added with this PR.
Previously there was no solution that would allow this with the expo-notification library. It was possible on iOS but not Android. 

# How

The daily interval uses the `TimeIntervalTrigger` on Android with a modified subsequent day trigger interval and on iOS it uses UNCalendarNotificationTrigger.

# Test Plan

Added two integration tests but those tests only verify first day event fire. We may want to remove these are they both take over 1 minute to run. If there are better solutions to verify the notification was scheduled properly, please let me know in the comments. 

**Note: I haven't updated the docs or provided a changelog. I wanted to first here if you are open to merging this PR. Thanks!**

### CODE REVIEW

File: apps/test-suite/tests/NewNotifications.js [hunk 1]
```diff
@@ -943,9 +943,61 @@ export async function test(t) {
           },
           16000
         );
+
+        t.it(
```
Comment (sjchmiela) [inline apps/test-suite/tests/NewNotifications.js:947]: Re:

> Added two integration tests but those tests only verify first day event fire. We may want to remove these are they both take over 1 minute to run. If there are better solutions to verify the notification was scheduled properly, please let me know in the comments.

I think we get away by:
- writing an automatic test that uses `getScheduledNotificationsAsync` to verify that the notification has been scheduled properly with the expected `hour` and `minute` params
- making these **real** tests (scheduling and waiting for the trigger) optional (triggerable with a user interaction, like [here](https://github.com/expo/expo/blob/385e7e0c54594aaf826b3266ea2ecc78a1948d5a/apps/test-suite/tests/NewNotifications.js#L1142-L1145))

**Review by sjchmiela — COMMENTED** [review]
Thank you so much for this contribution, we're flattered! 😍 This solves a real and useful scenario for scheduling notifications, we're looking forward to merging this feature!

I left a couple of (I hope) helpful comments. 🙂

File: packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/scheduling/NotificationScheduler.java [hunk 2]
```diff
@@ -27,6 +28,7 @@
 
 public class NotificationScheduler extends ExportedModule {
   private final static String EXPORTED_NAME = "ExpoNotificationScheduler";
+  private final static long DAY_IN_SECONDS = 86400;
```
[the remark this answers is not in the document]
Comment (thorbenprimke) [reply packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/scheduling/NotificationScheduler.java:31]: Sounds good.

File: packages/expo-notifications/src/scheduleNotificationAsync.ts [hunk 3]
```diff
@@ -29,6 +29,12 @@ function parseTrigger(userFacingTrigger: NotificationTriggerInput): NativeNotifi
     return { type: 'date', timestamp: userFacingTrigger.getTime() };
   } else if (typeof userFacingTrigger === 'number') {
     return { type: 'date', timestamp: userFacingTrigger };
+  } else if ('hour' in userFacingTrigger && 'minute' in userFacingTrigger) {
+    return {
+      type: 'dailyInterval',
+      hour: userFacingTrigger.hour ?? 0,
```
[the remark this answers is not in the document]
Comment (thorbenprimke) [reply packages/expo-notifications/src/scheduleNotificationAsync.ts:35]: Yes, I think it would be good to move the input validation logic here and raise an error if it's not within the bounds. That way it will be consistent and doesn't have to be duplicated on the native sides.

File: packages/expo-notifications/src/scheduleNotificationAsync.ts [hunk 4]
```diff
@@ -29,6 +29,12 @@ function parseTrigger(userFacingTrigger: NotificationTriggerInput): NativeNotifi
     return { type: 'date', timestamp: userFacingTrigger.getTime() };
   } else if (typeof userFacingTrigger === 'number') {
     return { type: 'date', timestamp: userFacingTrigger };
+  } else if ('hour' in userFacingTrigger && 'minute' in userFacingTrigger) {
```
[the remark this answers is not in the document]
Comment (thorbenprimke) [reply packages/expo-notifications/src/scheduleNotificationAsync.ts:32]: Good point. Yea, I think that can be achieved by adding

  Object.keys(userFacingTrigger).length === 2

to the conditional statement.

File: packages/expo-notifications/src/Notifications.types.ts [hunk 5]
```diff
@@ -129,12 +129,17 @@ export interface TimeIntervalTriggerInput {
   repeats?: boolean;
   seconds: number;
 }
+export interface DailyIntervalTriggerInput {
```
[the remark this answers is not in the document]
Comment (thorbenprimke) [reply packages/expo-notifications/src/Notifications.types.ts:132]: Yes, certainly. `DailyTriggerInput` it is.

File: packages/expo-notifications/ios/EXNotifications/Notifications/Scheduling/EXNotificationSchedulerModule.m [hunk 6]
```diff
@@ -12,6 +12,11 @@
 
 static NSString * const intervalNotificationTriggerType = @"timeInterval";
 static NSString * const intervalNotificationTriggerIntervalKey = @"seconds";
+static NSString * const intervalNotificationTriggerRepeatIntervalKey = @"repeatIntervalSeconds";
```
[the remark this answers is not in the document]
Comment (thorbenprimke) [reply packages/expo-notifications/ios/EXNotifications/Notifications/Scheduling/EXNotificationSchedulerModule.m:15]: Oh wow. That is left over from my attempt to subclass `UNTimeIntervalNotificationTrigger` which didn't go very well. 

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia1.giphy.com%2Fmedia%2F3og0INyCmHlNylks9O%2Fgiphy.gif&token=facepalm" data-gitmeme-token="facepalm"></a>

File: packages/expo-notifications/ios/EXNotifications/Notifications/Scheduling/EXNotificationSchedulerModule.m [hunk 7]
```diff
@@ -110,6 +115,16 @@ - (UNNotificationTrigger *)triggerFromParams:(NSDictionary *)params
 
     return [UNTimeIntervalNotificationTrigger triggerWithTimeInterval:[date timeIntervalSinceNow]
                                                               repeats:NO];
+
+  } else if ([dailyIntervalNotificationTriggerType isEqualToString:triggerType]) {
+    NSNumber *hour = [params objectForKey:dailyIntervalNotificationTriggerHourKey verifyingClass:[NSNumber class]];
+    NSNumber *minute = [params objectForKey:dailyIntervalNotificationTriggerMinuteKey verifyingClass:[NSNumber class]];
+    NSDateComponents *dateComponents = [NSDateComponents new];
+    dateComponents.hour = [hour integerValue];
+    dateComponents.minute = [minute integerValue];
```
[the remark this answers is not in the document]
Comment (thorbenprimke) [reply packages/expo-notifications/ios/EXNotifications/Notifications/Scheduling/EXNotificationSchedulerModule.m:124]: Yes, addressed above. I think it will be best to handle this on the JS side as the valid ranges of hour and minute do not depend on the platform. 

Not validating the input makes this PR sure seem like 

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia2.giphy.com%2Fmedia%2F3DzwbT4oF52JG%2Fgiphy.gif&token=amateur" data-gitmeme-token="amateur"></a>

File: packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/triggers/TimeIntervalTrigger.java [hunk 8]
```diff
@@ -47,7 +50,8 @@ public Date nextTriggerDate() {
 
     if (mRepeats) {
       while (mTriggerDate.before(now)) {
-        mTriggerDate.setTime(mTriggerDate.getTime() + mTimeInterval * 1000);
+        long interval = mRepeatInterval > 0 ? mRepeatInterval : mTimeInterval;
```
[the remark this answers is not in the document]
Comment (thorbenprimke) [reply packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/triggers/TimeIntervalTrigger.java:53]: Good point - it's only been two months of staying at home for covid and I already forgot about timezones. :D 

Yes, I think it makes sense to move the logic from `triggerFromParams` to a new class like `SchedulableNotificationTrigger` and just use it for the first and subsequent triggerDates.

File: apps/test-suite/tests/NewNotifications.js (hunk 1 above)
Comment (thorbenprimke) [reply apps/test-suite/tests/NewNotifications.js:947]: Sounds good.

File: apps/test-suite/tests/NewNotifications.js [hunk 9]
```diff
@@ -943,9 +943,61 @@ export async function test(t) {
...
+            await waitFor(60000);
+            t.expect(timesSpyHasBeenCalled).toBe(1);
+            subscription.remove();
+          },
+          70000
+        );
       }
 
       if (Platform.OS === 'ios') {
+        t.it(
```
[the remark this answers is not in the document]
Comment (thorbenprimke) [reply apps/test-suite/tests/NewNotifications.js:972]: There is an iOS and Android version of this test because Android uses 

  `addNotificationReceivedListener `

and iOS uses 

   `setNotificationHandler ` 

to verify that the notification is triggered. If I recall correctly, iOS didn't work with `addNotificationReceivedListener`.

File: apps/test-suite/tests/NewNotifications.js [hunk 10]
```diff
@@ -918,6 +918,44 @@ export async function test(t) {
...
+              type: 'calendar',
+              class: 'UNCalendarNotificationTrigger',
+              repeats: true,
+              dateComponents: {
+                ...trigger,
+                timeZone: null,
+                isLeapMonth: false,
+                calendar: null,
+              },
+            });
```
Comment (thorbenprimke) [inline apps/test-suite/tests/NewNotifications.js:951]: Since we use the `UNCalendarNotificationTrigger` on iOS, the validation between Android and iOS differs but that is expected so while not ideal, should needs to be this way.

File: apps/test-suite/tests/NewNotifications.js [hunk 11]
```diff
@@ -1179,6 +1217,69 @@ export async function test(t) {
...
+        t.afterEach(() => {
+          Notifications.setNotificationHandler(null);
+          Notifications.cancelAllScheduledNotificationsAsync();
+        });
+
+        t.it(
+          'triggers a repeating daily notification. only first event is verified.',
+          async () => {
+            const dateIn1Mimutes = new Date(
+              new Date().getTime() + (Platform.OS === 'ios' ? 120000 : 60000)
```
Comment (thorbenprimke) [inline apps/test-suite/tests/NewNotifications.js:1254]: On iOS because we are using the calendar with repeat, it needs to be greater than 60 seconds.

File: apps/test-suite/tests/NewNotifications.js [hunk 12]
```diff
@@ -1179,6 +1217,69 @@ export async function test(t) {
...
+            scheduledTime.setMilliseconds(0);
+            if (Platform.OS === 'android') {
+              scheduledTime.setMinutes(result[0].trigger.minute);
+            } else if (Platform.OS === 'ios') {
+              scheduledTime.setMinutes(result[0].trigger.dateComponents.minute);
+            } else {
+              throw new Error('Test does not support platfrom');
+            }
+            const miliSecondsToWait = scheduledTime - new Date().getTime() + 2000;
+            await waitFor(miliSecondsToWait);
```
Comment (thorbenprimke) [inline apps/test-suite/tests/NewNotifications.js:1276]: I'm looking up the scheduled time here in order to reduce the wait time here to a minimum.

File: packages/expo-notifications/src/scheduleNotificationAsync.ts [hunk 13]
```diff
@@ -29,14 +29,42 @@ function parseTrigger(userFacingTrigger: NotificationTriggerInput): NativeNotifi
...
+      type: 'daily',
+      hour,
+      minute,
+    };
+  } else if (
+    (Object.keys(userFacingTrigger).length === 1 && 'seconds' in userFacingTrigger) ||
+    (Object.keys(userFacingTrigger).length === 2 &&
+      'seconds' in userFacingTrigger &&
+      'repeats' in userFacingTrigger)
+  ) {
```
Comment (thorbenprimke) [inline packages/expo-notifications/src/scheduleNotificationAsync.ts:58]: The unit tests exposed that a calendar trigger with seconds would get incorrectly identified as a timeInterval.

File: packages/expo-notifications/src/scheduleNotificationAsync.ts [hunk 14]
```diff
@@ -29,14 +29,42 @@ function parseTrigger(userFacingTrigger: NotificationTriggerInput): NativeNotifi
...
     return {
       type: 'timeInterval',
       seconds: userFacingTrigger.seconds,
       repeats: userFacingTrigger.repeats ?? false,
     };
   } else {
+    // @ts-ignore
     const { repeats, ...calendarTrigger } = userFacingTrigger;
+    // @ts-ignore
     return { type: 'calendar', value: calendarTrigger, repeats };
```
Comment (thorbenprimke) [inline packages/expo-notifications/src/scheduleNotificationAsync.ts:68]: TS lint noted that DailyTrigger for example doesn't have `repeats`. I could rewrite this so that it only sets it conditionally but opted for ts-ignore instead. Let me know what you think about it.

File: packages/jest-expo/src/preset/expoModules.js [hunk 15]
```diff
@@ -380,6 +380,12 @@ module.exports = {
           { key: 1, argumentsCount: 0, name: 'getMacAddressAsync' },
           { key: 2, argumentsCount: 0, name: 'getNetworkStateAsync' },
         ],
+        ExpoNotificationScheduler: [
+          { key: 0, argumentsCount: 0, name: 'getAllScheduledNotificationsAsync' },
+          { key: 1, argumentsCount: 3, name: 'scheduleNotificationAsync' },
+          { key: 2, argumentsCount: 1, name: 'cancelScheduledNotificationAsync' },
+          { key: 3, argumentsCount: 0, name: 'cancelAllScheduledNotificationsAsync' },
+        ],
```
Comment (thorbenprimke) [inline packages/jest-expo/src/preset/expoModules.js:388]: required for the jest unit tests.

**Review by thorbenprimke — COMMENTED** [review]
@sjchmiela - I went through and changed it to `DailyTriggerInput`. 

I added some unit tests and integrations tests (automated and interactive) to verify this will work as expected. 

Let me know what you think. If all looks good, I'll add the documentation. 

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia2.giphy.com%2Fmedia%2F26u4lOMA8JKSnL9Uk%2Fgiphy.gif&token=done" data-gitmeme-token="done"></a>

**Comment by thorbenprimke** [comment]
@sjchmiela - I saw you dropped a few emojis. It's rebased now. Let me know if you have any other feedback or if it's good to go.

File: packages/expo-notifications/src/scheduleNotificationAsync.ts (hunk 14 above)
Comment (sjchmiela) [reply packages/expo-notifications/src/scheduleNotificationAsync.ts:68]: Yeah, I think ts-ignoring it makes sense for now 👍 (unless covering all other `seconds` cases earlier makes TS infer that down here the only possibility is for the `userFacingTrigger` to be an instance of `CalendarTriggerInput` which is the case now).

**Review by sjchmiela — COMMENTED** [review]
Thank you for adding the tests and everything! I only have a couple of small questions and typo-type nitpicks. Please note

File: apps/test-suite/tests/NewNotifications.js [hunk 16]
```diff
@@ -918,6 +918,44 @@ export async function test(t) {
         10000
       );
 
+      t.it(
+        'schedules a repeating daily notification. only first scheduled event is verified.',
+        async () => {
+          const dateNow = new Date();
+          const trigger = {
+            hour: dateNow.getHours(),
+            minute: dateNow.getMinutes() + 2,
```
[the remark this answers is not in the document]
Comment (thorbenprimke) [reply apps/test-suite/tests/NewNotifications.js:927]: Hmm. I think I'll still raise an error if `0` happens to be within a minute of the current time. 
I think I can just do `% 60` to ensure it's always a valid minute value.

File: apps/test-suite/tests/NewNotifications.js [hunk 17]
```diff
@@ -1179,6 +1217,69 @@ export async function test(t) {
...
+        const notification = {
+          title: 'Scheduled notification',
+          data: { key: 'value' },
+          badge: 2,
+          vibrate: [100, 100, 100, 100, 100, 100],
+          color: '#FF0000',
+        };
+
+        t.beforeEach(async () => {
+          Notifications.setNotificationHandler({
```
[the remark this answers is not in the document]
Comment (thorbenprimke) [reply apps/test-suite/tests/NewNotifications.js:1235]: Good point.

File: apps/test-suite/tests/NewNotifications.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (thorbenprimke) [reply apps/test-suite/tests/NewNotifications.js:1254]: done

File: apps/test-suite/tests/NewNotifications.js [hunk 18]
```diff
@@ -1179,6 +1217,69 @@ export async function test(t) {
...
+
+        t.afterEach(() => {
+          Notifications.setNotificationHandler(null);
+          Notifications.cancelAllScheduledNotificationsAsync();
+        });
+
+        t.it(
+          'triggers a repeating daily notification. only first event is verified.',
+          async () => {
+            const dateIn1Mimutes = new Date(
```
[the remark this answers is not in the document]
Comment (thorbenprimke) [reply apps/test-suite/tests/NewNotifications.js:1253]: `triggerDate` since it's not in 1 minute. :D

File: apps/test-suite/tests/NewNotifications.js [hunk 19]
```diff
@@ -1179,6 +1217,69 @@ export async function test(t) {
...
+                hour: dateIn1Mimutes.getHours(),
+                minute: dateIn1Mimutes.getMinutes(),
+              },
+            });
+            const result = await Notifications.getAllScheduledNotificationsAsync();
+            const scheduledTime = new Date();
+            scheduledTime.setSeconds(0);
+            scheduledTime.setMilliseconds(0);
+            if (Platform.OS === 'android') {
+              scheduledTime.setMinutes(result[0].trigger.minute);
```
[the remark this answers is not in the document]
Comment (thorbenprimke) [reply apps/test-suite/tests/NewNotifications.js:1269]: Yea, we can just use the `triggerDate` to init the `scheduledTime`.

File: apps/test-suite/tests/NewNotifications.js [hunk 20]
```diff
@@ -1179,6 +1217,69 @@ export async function test(t) {
...
+            } else if (Platform.OS === 'ios') {
+              scheduledTime.setMinutes(result[0].trigger.dateComponents.minute);
+            } else {
+              throw new Error('Test does not support platfrom');
+            }
+            const miliSecondsToWait = scheduledTime - new Date().getTime() + 2000;
```
[the remark this answers is not in the document]
Comment (thorbenprimke) [reply apps/test-suite/tests/NewNotifications.js:1275]: done

File: packages/expo-notifications/src/scheduleNotificationAsync.ts (hunk 13 above)
Comment (thorbenprimke) [reply packages/expo-notifications/src/scheduleNotificationAsync.ts:58]: Oh I see. I did miss that. I removed the above as correcting the test takes care of it. 

Hmm. I don't think I can add a check / TypeError that you have in #8261 for the dailyTrigger. However there could be a conflict between having minute and hour set and expecting repeats to default to false. The parameters would have to be changed to `minutes` / `hours` in order to properly check for this but

**Comment by thorbenprimke** [comment]
@sjchmiela - what do you think of changing `hour` to `hourOfDay` to make it explicit. That way there can be no confusion with the calendar type.

**Comment by thorbenprimke** [comment]
@sjchmiela - mind taking another look? 

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia2.giphy.com%2Fmedia%2F1oJKxy0ypYpQbK4b5n%2Fgiphy-downsized-medium.gif&token=ping" data-gitmeme-token="ping"></a>

**Review by sjchmiela — COMMENTED** [review]
Sorry for the delay! 😞 

If that wouldn't be too much of a problem, would it be possible to go with `hour`? I feel like adding `hourOfDay`:
- may make the API less intuitive from certain points of views — why `hourOfDay` and `minute` and not `hourOfDay` and `minuteOfHour`? Why would `hourOfDay` be more related to `DailyTrigger` than `CalendarTrigger`?
- we can let TS infer the type properly with a type guard function and thus get rid of the `@ts-ignore`
- a thorough documentation (on my TODO list) should rule out any misunderstandings

I've added suggestions wherever I could that may make it easier to rename the property. 🙂 

Also, we'd be grateful if you could add a CHANGELOG entry! 💯

File: packages/expo-notifications/src/scheduleNotificationAsync.ts [hunk 21]
```diff
@@ -29,13 +29,37 @@ function parseTrigger(userFacingTrigger: NotificationTriggerInput): NativeNotifi
     return { type: 'date', timestamp: userFacingTrigger.getTime() };
   } else if (typeof userFacingTrigger === 'number') {
     return { type: 'date', timestamp: userFacingTrigger };
+  } else if (
+    Object.keys(userFacingTrigger).length === 2 &&
+    'hourOfDay' in userFacingTrigger &&
+    'minute' in userFacingTrigger
+  ) {
```
[the remark this answers is not in the document]
Comment (thorbenprimke) [reply packages/expo-notifications/src/scheduleNotificationAsync.ts:36]: Let me try this to remove the ts-ignore.

**Comment by thorbenprimke** [comment]
Sounds good @sjchmiela . Let me switch it back quickly to `hour`. I already shipped this in our app to production yesterday. :D

**Comment by thorbenprimke** [comment]
@sjchmiela - it's good to be reviewed again. 

The unit tests are passing but there is still the edge case that arrives from using `hour` which is that a `hour` and `minute` trigger is interpreted as a DailyTrigger (based on the previous logic as well as new type guard) but in theory could have been a CalendarTrigger (that was meant to not repeat) with the repeats parameter omitted.

File: packages/expo-notifications/src/scheduleNotificationAsync.ts [hunk 22]
```diff
@@ -29,13 +29,37 @@ function parseTrigger(userFacingTrigger: NotificationTriggerInput): NativeNotifi
...
+      minute,
+    };
   } else if ('seconds' in userFacingTrigger) {
     return {
       type: 'timeInterval',
       seconds: userFacingTrigger.seconds,
       repeats: userFacingTrigger.repeats ?? false,
     };
   } else {
+    // @ts-ignore
```
[the remark this answers is not in the document]
Comment (thorbenprimke) [reply packages/expo-notifications/src/scheduleNotificationAsync.ts]: Yes, after type guard / is check, the ts-ignore is no longer required. Yay!

**Review by sjchmiela — APPROVED** [review]
Do you think it would make sense to make users add the explicit `{ repeats: true }` to `DailyTriggerInput` so that both they and we are sure they want the notification to be repeated?

Apart from this I have no other comments, thank you very much for bearing with me through this difficult, but so needed task. 🎖️ 🙇  _Chapeau bas!_

File: packages/expo-notifications/src/Notifications.types.ts [hunk 23]
```diff
@@ -129,12 +136,17 @@ export interface TimeIntervalTriggerInput {
   repeats?: boolean;
   seconds: number;
 }
+export interface DailyTriggerInput {
+  hour: number;
+  minute: number;
```
[the remark this answers is not in the document]
Comment (thorbenprimke) [reply packages/expo-notifications/src/Notifications.types.ts:141]: Okay. I like the addition of `repeats` as in that it fixes the ambiguity. However it does seem repetitive as daily (in `DailyTriggerInput`) I think implies repeating.

File: packages/expo-notifications/src/Notifications.types.ts [hunk 24]
```diff
@@ -70,6 +70,12 @@ export interface TimeIntervalNotificationTrigger {
   seconds: number;
 }
 
+export interface DailyNotificationTrigger {
+  type: 'daily';
+  hour: number;
+  minute: number;
```
[the remark this answers is not in the document]
Comment (thorbenprimke) [reply packages/expo-notifications/src/Notifications.types.ts:76]: I don't think we have to add `repeats` here as there shouldn't be any ambiguity in the `NotificationTrigger` handling as the Calendar one is for iOS only and the Daily one is for Android only.

File: packages/expo-notifications/src/scheduleNotificationAsync.ts [hunk 25]
```diff
@@ -29,6 +40,23 @@ function parseTrigger(userFacingTrigger: NotificationTriggerInput): NativeNotifi
     return { type: 'date', timestamp: userFacingTrigger.getTime() };
   } else if (typeof userFacingTrigger === 'number') {
     return { type: 'date', timestamp: userFacingTrigger };
+  } else if (isDailyTriggerInput(userFacingTrigger)) {
+    const hour = userFacingTrigger.hour;
+    const minute = userFacingTrigger.minute;
+    if (hour === undefined || minute === undefined) {
```
[the remark this answers is not in the document]
Comment (thorbenprimke) [reply packages/expo-notifications/src/scheduleNotificationAsync.ts:46]: Yea, I think that may have been an oversight / incorrect thinking. I was thinking the `null` case would fail the range check below but `undefined` would be zero. 

The `in` check only checks for the key. So I would think that having `undefined` or `null` for the value of a key that is being checked woudl still be true. 

Let me add `null` to the conditional as well.

File: packages/expo-notifications/src/Notifications.types.ts (hunk 24 above)
[the remark this answers is not in the document]
Comment (sjchmiela) [reply packages/expo-notifications/src/Notifications.types.ts:76]: True, I was torn between adding it and not (in case we ever added support for non-repeating daily trigger), but:
- going from `repeats: true` to `repeats: boolean` would be a breaking change even so
- I'm not sure if adding `repeats: boolean` now would make sense

I'll leave it up to you whether to add it or not. 🙂 (Let's remove it!)

**Comment by thorbenprimke** [comment]
@sjchmiela - What do you think about adding `repeats` only to `DailyTriggerInput` to strongly distinguish / identify the type but not adding it to `DailyNotificationTrigger` as it is not needed on the notification side?

**Comment by sjchmiela** [comment]
Let's do it that way! (as noted in https://github.com/expo/expo/pull/8199#discussion_r428117269 I wasn't sure whether to add it or not, so your suggestion is important here 👍 )

**Comment by thorbenprimke** [comment]
@sjchmiela - back on your plate. 

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia2.giphy.com%2Fmedia%2F2olWl39PDPSNi%2Fgiphy.gif&token=plate" data-gitmeme-token="plate"></a>

**Comment by thecodearrow** [comment]
Hi, the daily notifications trigger is working correctly on Day 1 in Android but fails on subsequent days. It gets triggered only once. Here's my code, please help me out ASAP. I have users complaining. 

 Notifications.scheduleNotificationAsync({
            content: {
                title: "Title",
                body: "Message",
            },
            trigger: {
                hour: hours,
                minute: minutes,
                repeats: true,
                priority: 'max'
            },
        });

**Comment by sjchmiela** [comment]
Hey @thecodearrow, is it possible that the issue is actually https://github.com/expo/expo/issues/9037? If not, please open a new one, it'll be easier for us to track than a comment on a PR. Thank you!

**Comment by thecodearrow** [comment]
@sjchmiela I tested it for shorter durations. It's working with no issues even after killing the app. 
Is the code that I have used correct for daily notification scheduling?

**Comment by RodolfoGS** [comment]
@thecodearrow please test the example that I added to https://github.com/expo/expo/issues/9037 to know if you also have that bug, maybe can be related, the notifications don't work when the device is idle (after 5 mins) with the new notifications api.
